File: Engr-T.stark/intervalListIntersection.cpp
vector<vector<int>> intervalIntersection(vector<vector<int>>& firstList, vector<vector<int>>& secondList) {
 
        vector<vector<int>> ans;
        int fN=firstList.size();
        int sN=secondList.size();
        
        int i=0,j=0;
        
        
        while(i<fN and j<sN){
            
            // Case 2
            if(max(firstList[i][0],secondList[j][0])<=min(firstList[i][1],secondList[j][1])){
                ans.push_back({max(firstList[i][0],secondList[j][0]),min(firstList[i][1],secondList[j][1])});
            }
            
            if(firstList[i][1]>=secondList[j][1]){
                j++;
            }
            else if(firstList[i][1]<secondList[j][1]){
                i++;
            }
            
        }
        return ans;
        
    }

File: olix/day-38/interval-list-intersection.js
/**
 * @param {number[][]} firstList
 * @param {number[][]} secondList
 * @return {number[][]}
 */
let intervalIntersection = function (firstList, secondList) {
  let result = [];
  let i = 0;
  j = 0;

  while (i < firstList.length && j < secondList.length) {
    let first = firstList[i];
    let second = secondList[j];
    if (first[1] < second[0]) {
      i++;
    } else if (second[1] < first[0]) {
      j++;
    } else {
      result.push([
        Math.max(first[0], second[0]),
        Math.min(first[1], second[1]),
      ]);
      if (first[1] < second[1]) {
        i++;
      } else {
        j++;
      }
    }
  }
  return result;
};
